CodeModel/CaDETModel/CodeItems/CaDETClass.cs

```diff
@@ -45,6 +45,20 @@ public List<CaDETClass> GetMethodLinkedVariableTypes()
                 .SelectMany(v => v.GetLinkedTypes()).ToList();
         }
 
+        public List<CaDETClass> GetMethodLinkedParameterTypes()
+        {
+            List<CaDETParameter> parameters = Members.SelectMany(m => m.Params).ToList();
+            return parameters.Select(p => p.Type)
+                   .Where(v => v.LinkedTypes != null)
+                   .SelectMany(v => v.LinkedTypes).ToList();
+        }
+
+        public List<CaDETClass> GetMethodInvocationsTypes()
+        {
+            List<CaDETMember> invokedMethods = Members.SelectMany(m => m.InvokedMethods).Where(m => m.Type is CaDETMemberType.Method).ToList();
```

Comment:
Invoked methods are already only methods (which is why we don't call them invoked members).